lib/src/lib/button/directives/preventDoubleClick.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, EventEmitter, HostListener, Output } from '@angular/core';
+import { Subject, timer } from 'rxjs';
+import { throttle } from 'rxjs/operators';
+
+@Directive({
+  selector: '[lab900PreventDoubleClick]',
+  standalone: true,
+})
+export class PreventDoubleClickDirective {
+  private clicks = new Subject<Event>();
+  private isThrottled = false;
+
+  @Output() public throttledClick = new EventEmitter<Event>();
+
+  public constructor() {
+    this.clicks.pipe(throttle(() => timer(1000))).subscribe((event) => {
+      this.throttledClick.emit(event);
+      this.isThrottled = false;
+    });
+  }
+
+  @HostListener('click', ['$event'])
```

Comment:
Hostlisteners are nice and easy to use, but they can have a performace pitfall.
You can read about it here https://blog.simplified.courses/the-danger-of-angular-host-listeners/
I'm just telling you this because it's good to know. 
You actually did it correctly here! :)